sklearn/_engine/base.py

```diff
@@ -91,16 +97,22 @@ def _get_engine_classes(engine_name, provider_names, engine_specs, default):
 
 def get_engine_classes(engine_name, default, verbose=False):
     provider_names = get_config()["engine_provider"]
+    # Single provider name was passed in
     if isinstance(provider_names, str):
         provider_names = (provider_names,)
-    elif not isinstance(provider_names, tuple):
-        # Make sure the provider names are a tuple to make it possible for the
-        # lru cache to hash them.
-        provider_names = tuple(provider_names)
+    # Single provider class was passed in
+    elif inspect.isclass(provider_names):
+        provider_names = (provider_names,)
```

Comment:
```suggestion
        # allow direct registration of engine classes to ease testing, debugging
        # and benchmarking without having to register a fake package with metadata
        # just to use a custom engine not meant to be used by end-users.
        provider_names = (provider_names,)
```

Reply:
Implemented this, but I moved the code to `_config.py`. Moved it because I didn't want to have to duplicate this logic in `base.py` and `_engine/base.py` which both access this config value.

What is the general thinking on having such "processing" logic in `_config.py`?